sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Util.scala

```diff
@@ -147,6 +137,25 @@ private[sql] object CatalogV2Util {
           replace(schema, update.fieldNames, field =>
             Some(field.withComment(update.newComment)))
 
+        case update: UpdateColumnPosition =>
+          def updateFieldPos(struct: StructType, name: String): StructType = {
+            val oldField = struct.fields.find(_.name == name).getOrElse {
+              throw new IllegalArgumentException("field not found: " + name)
+            }
+            val withFieldRemoved = StructType(struct.fields.filter(_ != oldField))
+            addField(withFieldRemoved, oldField, update.position())
+          }
+
+          update.fieldNames() match {
+            case Array(name) =>
+              updateFieldPos(schema, name)
+            case names =>
+              replace(schema, names.init, parent => parent.dataType match {
+                case parentType: StructType =>
```

Comment:
If the data type is not a struct, this should throw an exception, like the case for `UpdateColumnType`. Otherwise, this would throw `MatchError`, which isn't very helpful.

Reply:
done